universe/ValueRefs.cpp

```diff
@@ -249,6 +250,48 @@ namespace {
 }
 
 namespace ValueRef {
+
+template <>
+std::string ValueRef<std::string>::StringResult() const {
+    return Eval();
+}
+
+template <>
+std::string ValueRef<int>::StringResult() const {
+    return std::to_string(Eval());
+}
+
+template <>
+std::string ValueRef<float>::StringResult() const {
+    return std::to_string(Eval());
+}
+
+template <>
+std::string ValueRef<double>::StringResult() const {
+    return std::to_string(Eval());
+}
+
+template <>
+std::string ValueRef<std::vector<std::string>>::StringResult() const {
+    std::string s;
+    for (const auto &piece : Eval()) s += piece;
+    return s;
+}
+
+template <typename T>
+std::string ValueRef<T>::StringResult() const {
```

Comment:
... yeah, the default / templated implementation already does this, so do you need the specializations for types that are already supported by `to_string` ?

Reply:
probably you are right, i will ditch the specialisations for int,float,double and just trigger that the specialised cases get instantiated. will also try that in NamedValueRefManager probably i got the syntax wrong when i tried first.

Reply:
works